tests/chainer_tests/optimizers_tests/test_optimizers.py

```diff
@@ -46,6 +46,20 @@ def __call__(self, x):
         return F.sum((x - self.w) ** 2)
 
 
+class TestAllOptimizersCoverage(unittest.TestCase):
+    # Checks _all_optimizers covers all the built-in optimizers.
+
+    def test_all_optimizers_coverage(self):
+        module = chainer.optimizers
+        module_optimizers = []
+        for name in dir(module):
+            obj = getattr(module, name)
+            if (isinstance(obj, type) and issubclass(obj, chainer.Optimizer)):
+                module_optimizers.append(name)
+
+        assert sorted(_all_optimizers) == sorted(module_optimizers)
```

Comment:
Comparing `set` produces better error messages.
```
________________________________ TestA.test_set ________________________________

self = <test_tmp.TestA testMethod=test_set>

    def test_set(self):
>       assert set(self.x) == set(self.y)
E       AssertionError: assert {1, 2, 3, 4, 5} == {1, 2, 4, 5, 6}
E         Extra items in the left set:
E         3
E         Extra items in the right set:
E         6
E         Use -v to get the full diff

tests/test_tmp.py:11: AssertionError
____________________________ TestA.test_sorted_list ____________________________

self = <test_tmp.TestA testMethod=test_sorted_list>

    def test_sorted_list(self):
>       assert sorted(self.x) == sorted(self.y)
E       AssertionError: assert [1, 2, 3, 4, 5] == [1, 2, 4, 5, 6]
E         At index 2 diff: 3 != 4
E         Use -v to get the full diff

tests/test_tmp.py:13: AssertionError
```

Reply:
I wrote this way because converting to a set would eliminate duplicated items.
We could add another check `assert len(_all_optimizers) == len(set(_all_optimizers))`,
but I think the current form is better?

Reply:
Then, how about `collections.Counter(_all_optimizers)`?

Reply:
Hm, isn't that a bit overkill, and might obscure the intent? 🤔

Reply:
I am down for the current one or `assert len(_all_optimizers) == len(set(_all_optimizers))`
with an additional error message, although the test name and code is pretty self-explanatory🤔